samples/performance/texture_compression_basisu/texture_compression_basisu_tutorial.md

```diff
@@ -0,0 +1,239 @@
+-
+-     http://www.apache.org/licenses/LICENSE-2.0
+-
+- Unless required by applicable law or agreed to in writing, software
+- distributed under the License is distributed on an "AS IS" BASIS,
+- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+-->
+
+# Using Basis Universal supercompressed GPU texture codec with Vulkan
+
+## Overview
+
+This tutorial, along with the accompanying example code, demonstrates how to use Basis universal supercompressed GPU textures in a Vulkan application.
+
+## The KTX2 format
+[KTX](https://www.khronos.org/ktx/) is a GPU texture **container format** for storing different texture types (2D, cubemap, etc.) and texture formats (uncompressed and compressed). Version 2.0 added support for Basis Universal supercompressed textures.
+
+## Basis Universal
+
+[Basis Universal](https://github.com/BinomialLLC/basis_universal) is a supercompressed GPU texture data interchange system that implements the UASTC and ETC1S compressed formats that server as **transmission formats**. Both can be quickly transcoded to a wide variety of GPU native compressed an uncompressed formats like RGB/RGBA, PVRTC1, BCn, ETC1, ETC2, etc. This means that unlike a KTX2.0 file storing a BC3 textures, the data needs to be transcoded at runtime.
+
+![KTX and BasisU](./images/2021-ktx-universal-gpu-compressed-textures.png)
+
+## libktx
+
+This sample (as well as the repository) uses the *libktx* library from the [official Khronos KTX Software Repository](https://github.com/KhronosGroup/KTX-Software) for loading and transcoding the Basis Universal compressed KTX2.0 textures. It's included in the framework via this [CMakeLists.txt](../../../third_party/CMakeLists.txt) and also adds the Basis Universal transcoder:
+
+```CMake
+# libktx
+set(KTX_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ktx)
+
+set(KTX_SOURCES  
+    ...
+
+    # Basis Universal
+    ${KTX_DIR}/lib/basis_sgd.h
+    ${KTX_DIR}/lib/basis_transcode.cpp
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_containers.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_containers_impl.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_file_headers.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_global_selector_cb.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_global_selector_palette.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_transcoder_internal.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_transcoder_uastc.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_transcoder.cpp
+    ${KTX_DIR}/lib/basisu/transcoder/basisu_transcoder.h
+    ${KTX_DIR}/lib/basisu/transcoder/basisu.h
+    ${KTX_DIR}/lib/basisu/zstd/zstd.c
+
+    ...
+    # KTX2
+    ${KTX_DIR}/lib/texture2.c
+    ${KTX_DIR}/lib/texture2.h    
+```
+
+An alternative to including *libktx* via CMake would be using the pre-built binaries provided in the [KTX-Software repository](https://github.com/KhronosGroup/KTX-Software/releases).
```

Comment:
It is worth noting that basisu recently added support for a small footprint ktx2 loader. [How to configure and use the transcoder](https://github.com/BinomialLLC/basis_universal/wiki/How-to-Use-and-Configure-the-Transcoder) includes some pointers for implementers to get started.

Reply:
Any suggestions on wording? If I read that correctly, it would be an option to use basis_universal and it's transcoder instead of libktx for KTX2.0?

Reply:
Exactly.  You can get by with just the basisu library for KTX2 loading AND basisu transcoding.  The basisu library defines a `basist::ktx2_transcoder`, which you can initialize with a raw byte buffer that represents a ktx2 file and easily perform transcoding.  There's nothing wrong with using libktx, it's just another dependency that some projects may not need.

As for wording, perhaps something along the lines of:

> For projects just getting started with KTX2 and basisu, developers need only to rely on the Basis Universal dependency for loading KTX2 files and transcoding the basisu content, through the use of `basist::ktx2_transcoder`.  More information about how this is accomplished can be found in Binomial's documentation, [How to configure and use the transcoder](https://github.com/BinomialLLC/basis_universal/wiki/How-to-Use-and-Configure-the-Transcoder).

Reply:
Thanks for the wording :) I have added this with a slight change to the tutorial, so people will can see that this is an alternative to using libktx.